docs/internals/generic/_index.md

```diff
@@ -0,0 +1,215 @@
+
+- [(Dict[str, Any])](https://docs.python.org/3/library/stdtypes.html#typesmapping): Result of the executed command.
+
+```python {class="line-numbers linkable-line-numbers"}
+my_generic = Generic.from_robot(robot=robot, name="my_generic_service")
+
+raw_dict = {
+  "command": "raw",
+  "raw_input": "home"
+}
+await my_generic.do_command(raw_dict)
+```
+
+For more information, see the [Python SDK Docs](https://python.viam.dev/autoapi/viam/services/generic/client/index.html#viam.services.generic.client.GenericClient.do_command).
+
+{{% /tab %}}
+{{% tab name="Go" %}}
+
+**Parameters:**
+
+- `ctx` [(Context)](https://pkg.go.dev/context): A Context carries a deadline, a cancellation signal, and other values across API boundaries.
+- `cmd` [(map[string]interface{})](https://go.dev/blog/maps): The command to execute.
+
+**Returns:**
+
+- [(map[string]interface{})](https://go.dev/blog/maps): Result of the executed command.
+- [(error)](https://pkg.go.dev/builtin#error): An error, if one occurred.
+
+```go {class="line-numbers linkable-line-numbers"}
+myGeneric, err := generic.FromRobot(robot, "my_generic_service")
+
+resp, err := myGeneric.DoCommand(ctx, map[string]interface{}{"command": "example"})
+```
+
+For more information, see the [Go SDK Code](https://github.com/viamrobotics/api/blob/main/service/generic/v1/generic_grpc.pb.go).
+
+{{% /tab %}}
+{{% tab name="C++" %}}
+
+**Parameters:**
+
+- `command` [(string)](https://cplusplus.com/reference/string/string/): The command to execute.
+
+**Returns:**
+
+- [(string)](https://cplusplus.com/reference/string/string/): Result of the executed command.
+
+```cpp {class="line-numbers linkable-line-numbers"}
+result_map = client.do_command(command);
+```
+
+For more information, see the [C++ SDK Docs](https://cpp.viam.dev/classviam_1_1sdk_1_1GenericService.html)
+
+{{% /tab %}}
+{{< /tabs >}}
+
+### Close
+
+Safely shut down the resource and prevent further use.
+
+{{< tabs >}}
+{{% tab name="Python" %}}
+
+**Parameters:**
+
+- None
+
+**Returns:**
+
+- None
+
+```python {class="line-numbers linkable-line-numbers"}
+my_generic = Generic.from_robot(robot, "my_generic")
+
+await my_generic.close()
+```
+
+For more information, see the [Python SDK Docs](https://python.viam.dev/autoapi/viam/services/generic/client/index.html#viam.services.generic.client.GenericClient.close).
+
+{{% /tab %}}
+{{% tab name="Go" %}}
+
+**Parameters:**
+
+- `ctx` [(Context)](https://pkg.go.dev/context): A Context carries a deadline, a cancellation signal, and other values across API boundaries.
+
+**Returns:**
+
+- [(error)](https://pkg.go.dev/builtin#error) : An error, if one occurred. Close will never return an error for a generic resource.
+
+```go {class="line-numbers linkable-line-numbers"}
+myGeneric, err := generic.FromRobot(robot, "my_generic")
+
+err := myGeneric.Close(ctx)
+```
+
+For more information, see the [Go SDK Docs](https://pkg.go.dev/go.viam.com/rdk/resource#TriviallyCloseable).
+
+{{% /tab %}}
+{{< /tabs >}}
```

Comment:
It doesn't look like we need to explicitly close resources per the CPP examples I looked at. Can you confirm? Happy to add example here if you can point me to one.